subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoTaskExtension.java

```diff
@@ -52,11 +54,12 @@ public String getAsArg() {
         }
     }
 
+    private Project project;
```

Comment:
Create them differently. For example:

- Introduce a service that can be `@Injected` into any extension (or task or whatever).
- Use `@Inject` on the field and inject an instance directly.
- Add a new abstract type that extensions (or whatever) can extend, that provides these methods. 
- Use abstract getter and setter and let Gradle provide the instance.